src/java/org/apache/cassandra/io/sstable/SSTableIdentityIterator.java

```diff
@@ -138,7 +138,7 @@ public boolean hasNext()
         {
             return iterator.hasNext();
         }
-        catch (IndexOutOfBoundsException | VIntOutOfRangeException e)
+        catch (IndexOutOfBoundsException | VIntOutOfRangeException | AssertionError e)
```

Comment:
Why did you choose asserts instead of throwing a `CorruptSSTableException`?

Reply:
Lower level code cannot throw `CorruptSSTableException`, as it is a much-higher-level concept. Assertions are just one type of indication of unexpected data in the code (index out of bounds being another). The alternative to this is to go and change all data parsing to catch potential errors and translate them to `IOException`, which is not very feasible, and may be a problem if a piece of code can be used in other paths that don't involve reading from disk.